scripts/getting_started.sh

```diff
@@ -0,0 +1,583 @@
+    runCmd "make prepare-byoh-docker-host-image-dev" 0  "Making a byoh image: byoh-dev/node:v1.22.3 ..."
+    runCmd "cd -" 0
+else
+    echo "byoh image \"byoh-dev/node:v1.22.3\" existed."
+fi
+
+#check if byoh binary is existed
+byohBinaryFile="/tmp/byoh-hostagent-linux-amd64"
+if [ ! -f "${byohBinaryFile}" ]; then
+    runCmd "wget https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/releases/download/v0.1.0/byoh-hostagent-linux-amd64 -P /tmp" 0  "Downloading a byoh binary..."
+else
+    echo "${byohBinaryFile} existed."
+fi
+
+
+manageClusterConfFile="${HOME}/.kube/management-cluster.conf"
+if [ ! -f "${manageClusterConfFile}" ]; then
+    cp ${HOME}/.kube/config ${manageClusterConfFile}
+else
+    echo "${manageClusterConfFile} existed."
+fi
+
+KIND_IP=$(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' kind-control-plane) 
+
+grep ${KIND_IP} ${manageClusterConfFile} | grep -q 6443
+if [ $? -ne 0 ]; then
+    sed -i 's/    server\:.*/    server\: https\:\/\/'"${KIND_IP}"'\:6443/g' ${manageClusterConfFile}
+else
+    echo "Already modified ${manageClusterConfFile} before"
+fi
+
+
+## Register BYOH host to management cluster
+for i in {1..2}
+do
+  #Check if container "host$i" is already created.
+  docker ps -a | grep -q "host${i}"
+  if [ $? -ne 0 ]; then
+    runCmd "docker run --detach --tty --hostname host${i} --name host${i} --privileged --security-opt seccomp=unconfined --tmpfs /tmp --tmpfs /run --volume /var --volume /lib/modules:/lib/modules:ro --network kind byoh-dev/node:v1.22.3" 0 "Starting byoh container: host${i}..."
+    runByohAgent ${i} ${byohBinaryFile} ${manageClusterConfFile}
+  else
+    echo "Container \"host${i}\" is already created"
+
+    # check if docker status is valid
+    status=`docker container inspect -f '{{.State.Status}}' host${i}`
+    if [ "${status}" != "running" ]; then
+        echo "Error: Status of Container \"host${i}\" is ${status}, suggest to remove it, then rerun this script"
+        exit
+    fi
+
+    # check if byoh process running in this container
+    docker exec host${i} sh -c "ps aux" | grep -q byoh-hostagent
+    if [ $? -ne 0 ]; then
+        runByohAgent ${i} ${byohBinaryFile} ${manageClusterConfFile}
+    else
+        echo "byoh process is already ran in this container"
+    fi
+  fi
+done
+
+# Find a available IP for control plane endpoint
+CONTROL_PLANE_ENDPOINT_IP=""
+calcControlPlaneIP
+if [ -z "${CONTROL_PLANE_ENDPOINT_IP}" ]; then
+    echo "Can't get an available IP for control plane endpoint, exit...."
+    exit
+fi
+echo "CONTROL_PLANE_ENDPOINT_IP is ${CONTROL_PLANE_ENDPOINT_IP}"
+
+clusterYamlFile="/tmp/cluster-yaml"
+CONTROL_PLANE_ENDPOINT_IP=${CONTROL_PLANE_ENDPOINT_IP} clusterctl generate cluster byoh-cluster --infrastructure byoh --kubernetes-version v1.22.3 --control-plane-machine-count 1  --worker-machine-count 1 --flavor docker > "${clusterYamlFile}"
```

Comment:
I agree. But this is very much tightly coupled. Even if we want a different k8s version, we should change the script to support it.
We should take these values as args or env variables, but also define a default value. So that the user can use that if they wish different configurations.